src/services/dashboards/dashboards.svc.coffee

```diff
@@ -110,26 +110,47 @@ angular
             (success)->
               orgId = success.currentOrganization.id
 
+              # Retrieve dashboards
               $http.get(ImpacRoutes.dashboards.index(orgId)).then(
                 (dashboards)->
                   _self.setDashboards(dashboards.data).then(->
                     _self.setCurrentDashboard()
                     deferred.resolve(_self.config)
                     $log.info("Impac! - DashboardsSvc: loaded (force=#{force})")
 
-                  ).finally( -> _self.loadLocked=false )
+                  ).finally( -> _self.loadLocked = false )
                 (error)->
                   $log.error("Impac! - DashboardsSvc: cannot retrieve dashboards list for org: #{orgId}")
                   _self.loadLocked=false
                   deferred.reject(error)
               )
+
+              # Retrieve widgets templates
+              url = 
+                if ImpacTheming.get().dhbWidgetsConfig.templates.defaultToFinancialYear
+                  fy_end_month = ImpacMainSvc.getFinancialYearEndMonth()
+                  "#{ImpacRoutes.widgets.templates()}?financial_year_end_month=#{fy_end_month}"
+                else
+                  ImpacRoutes.widgets.templates()
+
+              $http.get(url).then(
+                (response) ->
+                  _self.setWidgetsTemplates(response.data.widgets)
+                  _self.loadLocked = false
+
```

Comment:
Very good point. I've refactored it to use $q.all and added a few comments.
Url is also now retrieved using a different function for more clarity.
![dashboard-loading](https://cloud.githubusercontent.com/assets/9582657/22677756/472f5042-ed49-11e6-8f2d-e8781bd26f2e.png)
On the screenshot above, we can see that `/widgets` and `/dashboards` are called at the same time, and the widgets and kpis start loading only after the two calls return.

Btw... `/widgets` is 11 times faster than `/dashboard` through `Her`... :)